Hardware/CMockHardwareMap.cpp

```diff
@@ -0,0 +1,350 @@
+{
+    // TODO: This sets power parameters for the model.
+    return 0;
+}
+
+float CMockHardwareMap::getHardPwmOutput(uint8_t channel)
+{
+    // TODO: this just returns what's been recorded for heater power
+    return 0;
+}
+
+#ifdef SOFT_PWM_OUTPUTS
+float CMockHardwareMap::setSoftPwmOutput(float power, uint8_t channel) = 0;
+float CMockHardwareMap::getSoftPwmOutput(uint8_t channel) = 0;
+#endif
+
+void CMockHardwareMap::setBreathingLight(float duty_cycle) {}
+
+void CMockHardwareMap::enableControlPower(bool b_enable)
+{
+    // todo: this should probably participate in the model of control current
+    // flow and temperature control.
+}
+
+void CMockHardwareMap::run()
+{
+    float total_radiator_flow = 0;
+    for (int i = 0; i < HARD_PWM_OUTPUTS; i++)
+    {
+        float heater_flow;
+        float radiator_flow;
+        /**
+         * Update temperature of the heater stage. heater_flow is heat flowing
+         * from heater to radiator due to temperature gradient. From heater
+         * point of view this is net loss of heat.
+         * The heater temperature is then updated based on heater power input
+         * and heater power loss to radiator. */
+        heater_flow = (m_temperature[i][0] - m_temperature[i][1]) /
+                      m_heat_conductance[i][0];
+        m_temperature[i][0] +=
+            (m_heater_power[i] * m_heater_rating[i] - heater_flow) *
+            m_run_period_ms / 1000 / m_heat_capacity[i][0];
+
+        /**
+         * Update temperature of the radiator stage. radiator_flow is heat
+         * flowing from radiator into incubator ambient air. From radiator point
+         * of view this is net loss of heat.
+         * The radiator temperature is then updated based on positive flow from
+         * heater and loss of heat to incubator.
+         * Total radiator heat flow is also accumulated to calculate incubator
+         * temperature. Negative sign is due to radiator loss being net positive
+         * flow for incubator.
+         */
+        radiator_flow = (m_temperature[i][1] - m_incubator_temperature) /
+                        m_heat_conductance[i][1];
+        m_temperature[i][1] += (heater_flow - radiator_flow) * m_run_period_ms /
+                               1000 / m_heat_capacity[i][1];
+        total_radiator_flow -= radiator_flow;
+    }
+    float incubator_flow;
+    incubator_flow =
+        (m_incubator_temperature - m_ambient_temperature) / m_incubator_loss;
+    m_incubator_temperature += (total_radiator_flow - incubator_flow) *
+                               m_run_period_ms / 1000 / m_incubator_capacity;
+}
+
+/**
+ * @note: This method is not for sending commands like >setHeaterPower(). This
+ * method is to simulate input from real world conditions changing as the device
+ * is being tested. For example: induce failure of a heater or thermistor. Or
+ * set different ambient temperature.
+ */
+bool CMockHardwareMap::newCommand(ICommand *p_command,
```

Comment:
Using C++ references will make the code more readable. 
If possible we should consider using references over pointers for safety and code readability reasons.